modular_citadel/code/game/machinery/wishgranter.dm

```diff
@@ -0,0 +1,107 @@
+				if("Power")	//Gives infinite power in exchange for infinite power going off in your face!
+					to_chat(user, "<B>Your wish is granted, but at a terrible cost...</B>")
+					to_chat(user, "The Wish Granter punishes you for your selfishness, warping itself into a delaminating supermatter shard!")
+					var/obj/item/stock_parts/cell/infinite/powah = new /obj/item/stock_parts/cell/infinite(get_turf(user))
+					if(user.put_in_hands(powah))
+						to_chat(user, "[powah] materializes into your hands!")
+					else
+						to_chat(user, "[powah] materializes onto the floor.")
+					var/obj/machinery/power/supermatter_shard/powerwish = new /obj/machinery/power/supermatter_shard(loc)
+					powerwish.damage = 700	//right at the emergency threshold
+					powerwish.produces_gas = FALSE
+					charges--
+					insisting = FALSE
+					if(!charges)
+						qdel(src)
+				if("Wealth")	//Gives 1 million space bucks in exchange for being turned into gold!
+					to_chat(user, "<B>Your wish is granted, but at a cost...</B>")
+					to_chat(user, "The Wish Granter punishes you for your selfishness, warping your body to match the greed in your heart.")
+					new /obj/structure/closet/crate/trashcart/moneywish(loc)
+					new /obj/structure/closet/crate/trashcart/moneywish(loc)
+					user.set_species(/datum/species/golem/gold)
+					charges--
+					insisting = FALSE
+					if(!charges)
+						qdel(src)
+				if("The Station To Disappear")	//teleports you to the station and makes you blind, making the station disappear for you!
+					to_chat(user, "<B>Your wish is 'granted', but at a terrible cost...</B>")
+					to_chat(user, "The Wish Granter punishes you for your selfishness, claiming your soul and warping your eyes to match the darkness in your heart.")
+					user.dna.add_mutation(BLINDMUT)
+					user.adjust_eye_damage(100)
+					var/list/destinations = list()
+					for(var/obj/item/device/radio/beacon/B in GLOB.teleportbeacons)
+						var/turf/T = get_turf(B)
+						if(is_station_level(T.z))
+							destinations += B
+					var/chosen_beacon = pick(destinations)
+					var/obj/effect/portal/jaunt_tunnel/J = new (get_turf(src), src, 100, null, FALSE, get_turf(chosen_beacon))
+					try_move_adjacent(J)
+					playsound(src,'sound/effects/sparks4.ogg',50,1)
+					charges--
+					insisting = FALSE
+					if(!charges)
+						qdel(src)
+				if("To Kill")	//Makes you kill things in exchange for rewards!
+					to_chat(user, "<B>Your wish is granted, but at a terrible cost...</B>")
+					to_chat(user, "The Wish Granter punishes you for your wickedness, warping itself into a dastardly creature for you to kill! ...but it almost seems to reward you for this.")
+					var/obj/item/melee/transforming/energy/sword/cx/killreward = new /obj/item/melee/transforming/energy/sword/cx(get_turf(user))
+					if(user.put_in_hands(killreward))
+						to_chat(user, "[killreward] materializes into your hands!")
+					else
+						to_chat(user, "[killreward] materializes onto the floor.")
```

Comment:
Could also add a 0.8 click delay multiplier for this wish.

Reply:
Addressed